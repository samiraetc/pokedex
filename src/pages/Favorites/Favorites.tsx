import React, { useState, useEffect } from 'react';
import Card from '../../components/Card/Card';
import { Pokemon } from '../../types/typing';
import { getFavoritesPokemon } from '../../utils/favorites';

export const COMPONENT_ID = 'pokedex-favorite';

const Favorites = () => {
    const [favorites, setFavorites] = useState([])
    const [pokemon, setPokemon] = useState<Pokemon>([] as any);

    useEffect(() => {
        setFavorites(getFavoritesPokemon(pokemon.id));
    }, [])

    return (
        <div id={COMPONENT_ID}>
            <div className="container">
                <div className="pokemon-list-container">
                    {favorites.map((pokemon: Pokemon) => {
                        return (
                            <Card pokemon={pokemon} />
                        )
                    })}
                </div>

            </div>
        </div>
    )
}

export default Favorites;