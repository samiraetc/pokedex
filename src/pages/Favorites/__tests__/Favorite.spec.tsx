import { render, configure, waitFor, getByText } from '@testing-library/react'
import Favorites, { COMPONENT_ID } from '../Favorites';

configure({
    testIdAttribute: 'id',
  });
  
describe('[FAVORITES]', () => {
    test('should render Favorites Page', async () => {
        const {asFragment, getByTestId, getByText} =  render(<Favorites/>);

        waitFor(() => {
            expect(getByText('bulbasaur'));
        })
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    test('should render Favorites Page and click to got to next page', async () => {
        const {asFragment, getByTestId, getByText} =  render(<Favorites/>);

        waitFor(() => {
            expect(getByText('bulbasaur'));
            expect(getByTestId('pokedex-pagination-next')).toBeInTheDocument();
        })

        expect(asFragment()).toMatchSnapshot();
    });

});