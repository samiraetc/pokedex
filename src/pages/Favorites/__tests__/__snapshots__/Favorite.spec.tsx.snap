// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FAVORITES] should render Favorites Page 1`] = `
<DocumentFragment>
  <div
    id="pokedex-favorite"
  >
    <div
      class="pokedex-loading"
      id="pokedex-loading"
    >
      <img
        alt="pokebola icon"
        class="pokedex-loading__icon"
        src="default"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`[FAVORITES] should render Favorites Page and click to got to next page 1`] = `
<DocumentFragment>
  <div
    id="pokedex-favorite"
  >
    <div
      class="pokedex-loading"
      id="pokedex-loading"
    >
      <img
        alt="pokebola icon"
        class="pokedex-loading__icon"
        src="default"
      />
    </div>
  </div>
</DocumentFragment>
`;
